### ISSUE TITLE
Finding or filtering with an empty string with a secondary index seems to be broken

### ISSUE THREAD

**Issue by nyh** [scylladb/scylla#8711]
The translated Cassandra unit test

* `cassandra_tests/validation/entities/secondary_index_test.py`::`testWithEmptyRestrictionValueAndSecondaryIndex`

fails in Scylla (and passes in Cassandra) in multiple test cases where an empty byte string is used in equality and inequality restrictions, where a secondary index is also involved. 

It seems that in equality checks against an empty byte string, rows that should match are missing, but in inequality (<, <=, etc.) checks, rows which should _not_ match do match.

I don't know if this bug is specific to the filtering **with a secondary index** (which this specific test exercises) or maybe exists also in filtering without secondary index - we should write a new test to check that. If the bug is not specific to secondary indexes, we should retag the issue.

**Comment by nyh** [comment]
This bug has already been fixed, it might have been a duplicate of #9375 which was also about indexing of empty strings.
The test listed still fails because it also depends on #4244, but I checked that all the parts which used to exercise *this* bug now pass.
